pkg/providers/vault/config.go

```diff
@@ -36,8 +38,18 @@ type SecretConfig struct {
 }
 
 // BuildURL constructs a URL which can be used to identify a HTTP based secret provider
-func (c SecretConfig) BuildURL() (path string) {
-	return fmt.Sprintf("%s://%s:%v%s", c.Protocol, c.Host, c.Port, c.Path)
+func (c SecretConfig) BuildURL(path string) (spURL string, err error) {
+	if _, err := url.Parse(path); err != nil {
```

Comment:
Seems to me like it would be more prudent to build the URL using string concatenation, parse it, then return then canonicalized url.String().  This will guarantee that the final URL is actually valid.  The code as written will allow the function to return https://edgex-vault:8200/http://github.com/edgexfoundry/go-mod-secrets/pull/51/files as the URL.

Reply:
makes sense

Reply:
well, it should not prevent it from returning URL in the query string, though. For example, there can be some case like https://someurl?redirect_url=http://github.com/edgexfoundry/... and that is a legitimate url...